### ISSUE TITLE
Comparing floats 

### ISSUE THREAD

**Issue by MrSaints** [syrusakbary/snapshottest#101]
Self-explanatory. 

Some potential ideas / handlers:

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertAlmostEqual
https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_almost_equal.html
https://docs.python.org/3/whatsnew/3.5.html#pep-485-a-function-for-testing-approximate-equality

**Comment by sbland** [comment]
Would this involve any more than creating a custom formatter in formatters.py and implementing it?

**Comment by MrSaints** [comment]
Yes -- we'll probably need a custom `assert_value_matches_snapshot` which relies on some of the methods mentioned above for approximate equality.

**Comment by medmunds** [comment]
You can register a custom formatter via the (undocumented) [`snapshottest.formatter.Formatter.register_formatter()`][register_formatter] static method, overriding the default [float formatter]. I believe you'd want to format floats into a new custom type (`FloatCloseTo(3.14159)`) that implements the desired `__eq__` along with a `__repr__` that can construct itself.

The built-in [file snapshot formatter] shows all the moving parts (plus a bunch more complexity you wouldn't need, around writing additional files).

Also (although I'm not proud of this suggestion), if all your floats are of similar precision, it *might* be sufficient to just serialize them as `Decimal`s?

[register_formatter]: https://github.com/syrusakbary/snapshottest/blob/0.6.0/snapshottest/formatter.py#L36
[float formatter]: https://github.com/syrusakbary/snapshottest/blob/0.6.0/snapshottest/formatters.py#L160
[file snapshot formatter]: https://github.com/syrusakbary/snapshottest/blob/0.6.0/snapshottest/file.py
